core/domain/topic_services.py

```diff
@@ -259,21 +259,20 @@ def apply_change_list(topic_id, change_list):
 
     Returns:
         dict, list(int), list(int). A dict with two fields: topic and
-            subtopic_pages containing the updated domain objects of each one,
-            a list of ids of the deleted subtopics and a list of ids of the
-            newly created subtopics.
+            modified_subtopic_pages containing the updated domain objects of
+            each one, a list of ids of the deleted subtopics and a list of ids
+            of the newly created subtopics.
     """
     topic = get_topic_by_id(topic_id)
-    subtopic_pages = subtopic_page_services.get_all_subtopic_pages_in_topic(
-        topic_id)
     newly_created_subtopic_ids = []
     deleted_subtopic_ids = []
+    modified_subtopic_pages = []
```

Comment:
If you don't filter out the no-ops beforehand, will you end up with gaps in your subtopic ID numbering scheme? If so, is that acceptable?

Might be better to filter beforehand and do a check here that no newly-added subtopic page is subsequently deleted.

Reply:
Will add the filters in the frontend, when it's done, as discussed.
But even so, I don't think such gaps would be an issue, as, if a user adds a subtopic, saves it and then deletes it in two different drafts, the gap would appear anyway.